Feat/match yocto wds config

alexhumphreys opened the pull request.

Commit:
Upgrade connman 1.30 -> 1.31

Commit:
Add Vitaly's config, with added CONFIG_LIBNL32=y

alexhumphreys commented on meta-jaspar/recipes-connectivity/wpa-supplicant/wpa-supplicant/defconfig:
```diff
@@ -516,9 +495,7 @@ CONFIG_CTRL_IFACE_DBUS_NEW=y
-CONFIG_AP=y
-
-CONFIG_BGSCAN_SIMPLE=y
+#CONFIG_AP=y
```
@cajun-rat Not sure, but this might be important...

cajun-rat commented:
We are going to sit on this until we can confirm that it fixes the WDS issues.  When we can get WDS working with it, we'll merge it but otherwise we are risking introducing bugs.

cajun-rat commented:
Am I right in thinking that we believe connman 1.31 breaks the streaming use case?

alexhumphreys commented:
I believe so, not sure what state my tmp/ folder is in but I can rebuild and test later. Maybe it was that CONFIG_AP thing afterall...

Commit:
Enable AP mode

Without CONFIG_AP, connman cannot do tethering mode

Commit:
Remove message about updating jaspar repo

Since we don't use Vagrant any more, keeping the two repos in sync is no longer
necessary.

Commit:
WIP

cajun-rat commented:
This seems to stop the Streaming mode working :(

To reproduce:

1) Boot the image
2) Click 'streaming'
3) ssh qemu connmanctl technologies

In a good build (e.g. master) it will say 'Tethering = True' under /net/connman/technology/wifi

With this, it will say 'Tethering = False', and typing 'tether wifi on' in connmanctl will result in an error like so:

```
connmanctl> tether wifi on
Error enabling wifi tethering: Not supported
```

stevana commented:
Why is this branch needed? It's untested and only seems to bump connman to 1.31 (which we haven't confirmed improves anything).

alexhumphreys commented:
@stevana it was opened 12 days ago. I think I was trying to match yocto as close as possible to either the vagrant or Vitaly's setup. If we're happy enough with master and if we don't need and/or can't fix connman 1.31, then I guess we don't need this branch.

stevana commented:
Good, since connman 1.31 already has its own branch; let's close this one.
